src/EA.Weee.Web/Areas/Producer/Mappings/ToViewModel/CheckAnswersMap.cs

```diff
@@ -46,10 +47,17 @@ public CheckAnswersViewModel Map(SmallProducerSubmissionMapperData source)
                 ServiceOfNoticeData = serviceOfNoticemodel,
                 RepresentingCompanyDetails = representingCompanyDetailsmodel,
                 EeeData = editEeeDatamodel,
-                ComplianceYear = submissionData.CurrentSubmission.ComplianceYear
+                ComplianceYear = year
             };
 
             return viewModel;
         }
+
+        public CheckAnswersViewModel Map(SmallProducerSubmissionMapperData source)
+        {
+            // Uses the current year if no explicit year is provided
+            int year = source.Year ?? DateTime.Now.Year;
```

Comment:
@will-hope98  think if the source year isn't set we could probably get away with using submissionData.CurrentSubmission.ComplianceYear